conans/test/integration/toolchains/cmake/cmakedeps/test_cmakedeps.py

```diff
@@ -707,3 +707,39 @@ def generate(self):
     assert 'set(dep_NO_SONAME_MODE_RELEASE TRUE)' in dep
     other = c.load("app/other-release-data.cmake")
     assert 'set(other_other_mycomp1_NO_SONAME_MODE_RELEASE TRUE)' in other
+
+def test_cmakedeps_set_legacy_variable_name():
```

Comment:
As we are altering some behaviour, I think it would be nice that this tests first that if you set the `cmake_file_name` property everything works as expected, but if you later set both `cmake_file_name` and `cmake_legacy_variable_prefix` the new one has preference.